gbdxtools/images/meta.py

```diff
@@ -135,7 +135,67 @@ def window_at(self, geom, x_size, y_size, no_padding=False):
         if minx < 0 or miny < 0 or maxx > x_max or maxy > y_max:
             raise ValueError("Input geometry resulted in a window outside of the image")
         return self[:, miny:maxy, minx:maxx]
-        
+
+    def window_cover(self, window_shape):
+        """
+        Returns a list of windows of a specified shape over an entire AOI.
+        args:
+            window_shape (tuple): The desired shape of each image as (height, width) in pixels.
+        returns:
+            imageWindow: a list of image tiles covering the entirety of the AOI
+        """
+        height,width=window_shape[0],window_shape[1]
+        _ndepth,_nheight,_nwidth = self.shape
+        nheight, _m = divmod(_nheight, height)
+        nwidth, _n = divmod(_nwidth, width)
+        if _m != 0 or _n != 0:
+            raise ValueError("Window dimensions invalid. Window dimensions must be divisible by the width and height of the image AOI.")
```

Comment:
I think this is the line in question at this point. But actually now i realize the other issue. What we really want here is to return a list of dasks that cover a geometry and are all of a given y/x size (shape). The method implemented here is basically just redoing dask's `rechunk` method but returning each chunk, which may in fact have a place as a method.

Reply:
Sorry, I am kind of confused by your wording on the second part of your comment. Can you clarify?